af/fastpath/fastpath/s3feeder.py

```diff
@@ -80,25 +82,85 @@ def load_multiple(fn: str) -> Generator[MsmtTup, None, None]:
             #     metrics.incr("yaml_normalization")
             #     yield (None, msm)
 
+    elif fn.endswith(".tar.gz"):
+        # minican with missing gzipping :(
+        tf = tarfile.open(fn)
+        while True:
+            m = tf.next()
+            if m is None:
+                # end of tarball
+                tf.close()
+                break
+            log.debug("Loading %s", m.name)
+            k = tf.extractfile(m)
+            assert k is not None
+            if not m.name.endswith(".post"):
+                log.error("Unexpected filename")
+                continue
+
+            try:
+                j = ujson.loads(k.read())
+            except Exception:
+                log.error(repr(k[:100]), exc_info=1)
+
+            fmt = j.get("format", "")
+            if fmt == "json":
+                msm = j.get("content", {})
+                yield (None, msm, None)
+
+            elif fmt == "yaml":
+                log.info("Skipping YAML")
+
+            else:
+                log.info("Ignoring invalid post")
+
     else:
-        raise RuntimeError(fn)
+        raise RuntimeError(f"Unexpected [mini]can filename '{fn}'")
 
 
 def create_s3_client():
     return boto3.client("s3", config=botoConfig(signature_version=botoSigUNSIGNED))
 
 
-def list_cans_on_s3_for_a_day(s3, day):
+def list_cans_on_s3_for_a_day(s3, day: date):
+    """List legacy cans."""
     prefix = f"{day}/"
-    r = s3.list_objects_v2(Bucket=BUCKET_NAME, Prefix="canned/" + prefix)
-    files = []
-    assert "Contents" in r
-    for filedesc in r["Contents"]:
-        fname = filedesc["Key"][7:]  # trim away "canned/"
-        files.append((fname, filedesc["Size"]))
+    r = s3.list_objects_v2(Bucket=CAN_BUCKET_NAME, Prefix="canned/" + prefix)
+
+    if ("Contents" in r) ^ (day <= date(2020, 10, 21)):
```

Comment:
Maybe this magic date can be defined as a constant since you use it in two places.

Reply:
Nevermind, I was confused by the date below, but I now realise they are different dates.